### PR TITLE
dev-util/valgrind: fix for --xml-socket option

### PR DESCRIPTION
**Pull request by winterheart**
Fix for Qt Creator working.

Closes: https://bugs.gentoo.org/641790
Package-Manager: Portage-2.3.13, Repoman-2.3.3

### CODE REVIEW

File: dev-util/valgrind/valgrind-3.13.0-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	# Don't use mpicc unless the user asked for it (bug #258832)
+	if ! use mpi; then
+		myconf="${myconf} --without-mpicc"
+	fi
+
+	econf ${myconf}
+}
+
+src_install() {
+	emake DESTDIR="${D}" install
```
Comment (SoapGentoo) [inline dev-util/valgrind/valgrind-3.13.0-r2.ebuild]: `default`

File: dev-util/valgrind/valgrind-3.13.0-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+	if [[ ${PV} == "9999" ]]; then
+		# Otherwise FAQ.txt won't exist:
+		emake -C docs FAQ.txt
+		mv docs/FAQ.txt . || die "Couldn't move FAQ.txt"
+	fi
+
+	dodoc AUTHORS FAQ.txt NEWS README*
```
Comment (SoapGentoo) [inline dev-util/valgrind/valgrind-3.13.0-r2.ebuild]: `dodoc FAQ.txt` (the rest is part of `einstalldocs`)

File: dev-util/valgrind/valgrind-3.13.0-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	filter-flags -fstack-protector-strong
+	filter-flags -m64 -mx32
+	replace-flags -ggdb3 -ggdb2
+
+	if use amd64 || use ppc64; then
+		! has_multilib_profile && myconf="${myconf} --enable-only64bit"
+	fi
+
+	# Force bitness on darwin, bug #306467
+	use x86-macos && myconf="${myconf} --enable-only32bit"
```
Comment (SoapGentoo) [inline dev-util/valgrind/valgrind-3.13.0-r2.ebuild]: make `myconf` a bash array instead of a whitespace list

**Comment by winterheart** [comment]
I'm authorized to make changes into ebuilds? 9999 requires these changes too. @blueness ?

**Comment by austin987** [comment]
I'm on holiday and won't have a good chance to look, but at first glance, the ggdb3 segfault needs a valgrind bug filed. The multiarch flags are also suspect, but in fairness, I haven't looked at the linked bug.

At a minimum, I want to see the ggdb3 bug filed upstream.

**Comment by winterheart** [comment]
@SoapGentoo done.

**Comment by austin987** [comment]
Committed as https://github.com/gentoo/gentoo/commit/727a3c7b1f8094d1a66314e64077fe93dbe5dd2d

**Pull request closed without merging**